sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala

```diff
@@ -66,6 +66,30 @@ private[execution] sealed trait HashedRelation extends KnownSizeEstimation {
     throw new UnsupportedOperationException
   }
 
+  /**
+   * Returns an iterator for key index and matched rows.
+   *
+   * Returns null if there is no matched rows.
+   */
+  def getWithKeyIndex(key: InternalRow): Iterator[ValueRowWithKeyIndex]
+
+  /**
+   * Returns key index and matched single row.
+   *
+   * Returns null if there is no matched rows.
+   */
+  def getValueWithKeyIndex(key: InternalRow): ValueRowWithKeyIndex
```

Comment:
Is this for unique key case only?

Reply:
@viirya - yes. Similar to definition of `def getValue(key: InternalRow): InternalRow` above. I added comment to say `Returns key index and matched single row.`, to be consistent with comment of `getValue`. Hope this is clear enough.

Reply:
Can you add comment saying this is for unique key case?

Reply:
@viirya - sure, added.